core-tests/shared/src/test/scala/zio/stm/TArraySpec.scala

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package zio.stm
+
+import zio.{ FiberFailure, UIO, ZIO, ZIOBaseSpec }
+import zio.test._
+import zio.test.Assertion._
+import TArraySpecUtil._
+
+object TArraySpec
+    extends ZIOBaseSpec(
+      suite("TArraySpec")(
+        suite("apply")(
+          testM("happy-path") {
+            for {
+              tArray <- makeTArray(1)(42)
+              value  <- tArray(0).commit
+            } yield assert(value, equalTo(42))
+          },
+          testM("dies with ArrayIndexOutOfBounds when index is out of bounds") {
+            for {
+              tArray <- makeTArray(1)(42)
+              result     <- ZIO.effect(tArray(-1)).run
+            } yield assert(result, fails(Assertion.assertion[Throwable]("isArrayIndexOutOfBoundsException")()(_.isInstanceOf[ArrayIndexOutOfBoundsException])))
```

Comment:
I see you could get away from using `dies` by making `result` a `ZIO[_, _, STM[_, _]]` - nice!
I think you can make a helper `val` for this assertion and use it in two other tests.

In sbt please run `fmt` to auto-format code.

Thank you very much for your work!